code/game/turfs/simulated/floor/asteroid.dm

```diff
@@ -31,46 +48,50 @@
 /turf/simulated/floor/plating/asteroid/remove_plating()
 	return
 
-/turf/simulated/floor/plating/asteroid/ex_act(severity, target)
+/turf/simulated/floor/plating/asteroid/ex_act(severity)
+	if(!can_dig())
+		return
 	switch(severity)
 		if(3)
 			return
 		if(2)
 			if(prob(20))
-				gets_dug()
+				getDug()
 		if(1)
-			gets_dug()
+			getDug()
 
-/turf/simulated/floor/plating/asteroid/attackby(obj/item/W, mob/user, params)
+/turf/simulated/floor/plating/asteroid/attackby(obj/item/I, mob/user, params)
 	//note that this proc does not call ..()
-	if(!W || !user)
-		return 0
+	if(!I|| !user)
+		return FALSE
+
+	if((istype(I, /obj/item/shovel) || istype(I, /obj/item/pickaxe)))
+		if(!can_dig(user))
+			return TRUE
 
-	if((istype(W, /obj/item/shovel) || istype(W, /obj/item/pickaxe)))
 		var/turf/T = get_turf(user)
 		if(!istype(T))
 			return
 
-		if(dug)
-			to_chat(user, "<span class='warning'>This area has already been dug!</span>")
-			return
-
 		to_chat(user, "<span class='notice'>You start digging...</span>")
-		playsound(src, W.usesound, 50, 1)
-		if(do_after(user, 20 * W.toolspeed, target = src))
+
+		playsound(src, I.usesound, 50, TRUE)
+		if(do_after(user, 40 * I.toolspeed, target = src))
```

Comment:
Technically it'll take longer to dig up sand now; there was an inconsistency here, for some reason; mining, regardless of type/location is supposed to have a modifier of `40`, not `20`